src/js/contexts/RootsContext.js

```diff
@@ -1,6 +1,21 @@
-import React from 'react';
+import React, { useContext, useMemo, useRef } from 'react';
 
 // When toggling aria-hidden values, we only want to affect elements
 // in the DOM that come from Grommet, so we track those elements in this
-// context value. See FocusedContainer.js
-export const RootsContext = React.createContext([]);
+// context value. We also only want `trapFocus` to apply to the most recently
+// opened Drop/Layer. See FocusedContainer.js
+export const RootsContext = React.createContext(undefined);
+
+export const useRoots = () => {
+  // If FocusedContainer is within Grommet React tree, there will be a value
+  // from RootsContext. If not, set up a default value that can be passed
+  // to drops that might open from within a drop.
+  const defaultRoots = useRef([]);
+  const defaultValue = useMemo(() => ({ roots: defaultRoots }), []);
+  const existingRoots = useContext(RootsContext);
+
+  return {
+    contextValue: existingRoots || defaultValue,
+    withinGrommet: existingRoots?.roots !== undefined,
```

Comment:
I'm wondering if this should be renamed to something like `isDefault`? Is it true in the "not within grommet" case that the first FocusedContainer will add the context but any children in that container that do a drop or Layer will result in useRoots returning `true` for `withinGrommet` and nicely not add another provider? (ie. `withinGrommet` is a little misleading.)

Reply:
Correct -- "any children in that container that do a drop or Layer will result in useRoots returning true nicely not add another provider". Open to changing the name of the variable-- agree in that case it's a bit misleading.

Reply:
Renamed to `hasRoots`